File: src/web/magmac/app/compile/rule/fold/StatementFolder.ts
export class StatementFolder {
	public fold( state : DivideState,  c : char) : DivideState { let appended : var=state.append( c);if(true){ return appended.advance( );;}if(true){ return appended.advance( ).exit( );;}if(true){ return appended.enter( );;}if(true){ return appended.exit( );;}return appended;;}
	public createDelimiter() : String {return "";;}
}

File: packages/compiler-java/src/main/java/magmac/app/compile/rule/fold/StatementFolder.java
package magmac.app.compile.rule.fold;

import magmac.app.compile.rule.divide.DivideState;

public class StatementFolder implements Folder {
    @Override
    public DivideState fold(DivideState state, char c) {
        var appended = state.append(c);
        if (';' == c && appended.isLevel()) {
            return appended.advance();
        }
        if ('}' == c && appended.isShallow()) {
            return appended.advance().exit();
        }
        if ('\n' == c && appended.isLevel()
                && !appended.inLineComment()
                && !appended.inBlockComment()
                && !appended.inSingle()
                && !appended.inDouble()) {
            return appended.advance();
        }
        if ('{' == c || '(' == c) {
            return appended.enter();
        }
        if ('}' == c || ')' == c) {
            return appended.exit();
        }
        return appended;
    }

    @Override
    public String createDelimiter() {
        return "";
    }
}
